build-complete-image.sh

```
@@ -0,0 +1,351 @@
+if [ ! -f "$UBOOT_FILES_DIR/idbloader.img" ] || [ ! -f "$UBOOT_FILES_DIR/uboot.img" ]; then
+    echo -e "${YELLOW}========================================${NC}"
+    echo -e "${YELLOW}Step 0: Creating dummy U-Boot files${NC}"
+    echo -e "${YELLOW}========================================${NC}"
+    echo -e "${YELLOW}(Required for bitbake to parse recipes)${NC}"
+    echo ""
+
+    # Create dummy files so bitbake can parse recipes
+    mkdir -p "$UBOOT_FILES_DIR"
+    dd if=/dev/zero of="$UBOOT_FILES_DIR/idbloader.img" bs=1K count=312 2>/dev/null
+    dd if=/dev/zero of="$UBOOT_FILES_DIR/uboot.img" bs=1M count=4 2>/dev/null
+    DUMMY_UBOOT_CREATED=1
```

Shellcheck (0.11.0)

[warning] DUMMY_UBOOT_CREATED appears unused. Verify use (or export if used externally).

(SC2034)